zero-waste-service/routes/bin.js

```diff
@@ -0,0 +1,58 @@
+const express = require('express')
+const router = express.Router();
+const models = require('../models/index')
+
+router.post('/', function (req, res) {
+    const userId = req.body.userId;
+    if(userId == null) {
+        throw new Error("Input format is incorrect")
+    }
+
+    const bin = models.bin;
+    bin.find()
+    .where("userId")
+    .equals(userId)
+    .select("name userId status")
+    .exec((err, result) => {
+        if (err) return "Error with fetching data";
+
+        return res.status(200).send(result);
+    })
+})
+
+router.post('/save', function (req, res) {
+    const name = req.body.name;
+    const pid = req.body.pid;
+    const userId = req.body.userId;
+    const status = req.body?.status ?? 'Fillable';
+
+    if(!pid || !name || !userId || !status) {
+        throw new Error("Input format is incorrect")
+    }
+
+    const bin = models.bin;
+
+    const newBin = new bin({ name, pid, userId, status })
+
+    bin.countDocuments({pid: pid, userId: userId}, function (err, count) {
```

Comment:
No I think it will cause a system inconsistency. For example, if one user does not remove a bin from the system and gives it to their neighbor who registers the bin to the system, then we get 2 bins with same pid in the system. Now, which bin should we refer to using pid if asked by a user? To prevent this issue, we need userId.

Reply:
Got it, thanks!